modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -177,6 +190,28 @@ object ConsensusManager {
             .handleErrorWith(logger.error(_)(s"Error checking for consensus state update {key=${key.show}}"))
         }.void
 
+      private def getPeerCandidates(peers: List[Peer]): F[List[Peer]] = {
+        case class PeerSnapshot(ordinal: SnapshotOrdinal, hash: Hash, peer: Peer)
+        val groupFn: PeerSnapshot => (SnapshotOrdinal, Hash) = v => (v.ordinal, v.hash)
+
+        for {
```

Comment:
You have a single flatMap, and then all the other assignments are just pure values. You don't need a for comprehension for that. Also, there is no need to specify each and every type within the for comprehension.

Reply:
Changed.